tests/api/api/test_artifacts.py

```diff
@@ -55,6 +56,63 @@ def _create_project(client: TestClient, project_name: str = PROJECT):
     return resp
 
 
+def test_store_artifact_with_invalid_key(db: Session, client: TestClient):
+    _create_project(client)
+    key_path = "some-key/with-slash"
+
+    # sanity, valid key path works
+    resp = client.post(
+        STORE_API_ARTIFACTS_PATH.format(project=PROJECT, uid=UID, key=KEY, tag=TAG),
+        data="{}",
+    )
+    assert resp.status_code == HTTPStatus.OK.value
+
+    # use invalid key path, expect its validation to fail
+    resp = client.post(
+        STORE_API_ARTIFACTS_PATH.format(
+            project=PROJECT, uid=UID, key=key_path, tag=TAG
+        ),
+        data="{}",
+    )
+    assert resp.status_code == HTTPStatus.BAD_REQUEST.value
+
+
+def test_store_artifact_backwards_compatibility(db: Session, client: TestClient):
+    _create_project(client)
+    key_path = "some-key/with-slash"
+    artifact = mlrun.artifacts.Artifact(key=key_path, body="123")
+    artifact2 = mlrun.artifacts.Artifact(key=key_path, body="1234")
+
```

Comment:
comment above each "test" step what you are doing, e.g.:

```python
# create X
do_something()

# ensure it exists
get_something()

# do something else
do_something_else()

# ensure it still there
get_something()
```
That way, it is more easy to read tests